frontend/online-shop/src/App.tsx

```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import "./App.css";
+import ProductList from "./components/ProductList";
+import ProductDetailedView from "./components/ProductDetailedView";
+import { CssBaseline, ThemeProvider } from "@material-ui/core";
+import StandardTheme from "./themes/StandardTheme";
+import { BrowserRouter, Redirect, Route, Switch } from "react-router-dom";
+import Cart from "./components/Cart";
+// import ProductItem from "./components/ProductItem";
+
+function App() {
+  return (
+    <ThemeProvider theme={StandardTheme}>
+      <CssBaseline>
+        <BrowserRouter>
+          <Switch>
+            <Route path="/products/:id" component={ProductDetailedView} />
+            <Route exact path="/">
+              <Redirect to="/products" />
+            </Route>
+            <Route path="/products" component={ProductList} />
+            <Route path="/cart" component={Cart} />
+            <Route path="/" render={() => <div>404: Page not found</div>} />
```

Comment:
This is a great thing that you wanted to achieve (the 404 page). The only observation here is that you need to use the `*` wildcard instead of that `/`. This way every route that is not identified in the above ones will enter here and return that 404 div. You could also create a 404 component.